File: 3rd Semester/Advanced Programming Methods(Done)/Laboratory/C#LabAssignment/sharplab/Model/Utils/FileTable.cs
using System;
using System.Collections;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace sharplab.Model.Utils
{
    class FileTable<T> : IFileTable<T>
    {
        private IDictionary<int, Tuple> fileT;
        private int idx;

        public FileTable()
        {
            this.fileT = new SortedDictionary<int, Tuple>();
            this.idx = 0;
        }

        public void Add(Tuple elem)
        {
            idx++;
            fileT.Add(idx, elem);
        }

        public int GetKey() { return idx; }

        public Tuple Get(int i) { return fileT[i]; }

        public void Remove(int key) { fileT.Remove(key); }

        public override string ToString()
        {
            String msg = "";
            foreach (int i in fileT.Keys.ToArray())
            {
                Tuple tup = this.Get(i);
                msg += i + " " + tup.GetName() + '\n';
            }
            return msg;
        }

        public IDictionary<int, Tuple> GetFT() { return fileT; }
    }
}

File: 3rd Semester/Advanced Programming Methods(Done)/Laboratory/src (BUX)/Model/Utils/FileTable.java
package Model.Utils;

import java.util.HashMap;
import java.util.Map;

public class FileTable<T> implements IFileTable<T>
{
    private Map<Integer, T> fileT;
    private int idx;

    public FileTable()
    {
        this.fileT = new HashMap<>();
        this.idx = 0;
    }

    public void add(T elem) {
        idx++;
        fileT.put(idx, elem);
    }

    public int getKey() {return idx;}

    public T get(int i) {return fileT.get(i);}

    public void remove(int key) {fileT.remove(key);}

    public String toString() {
        String msg = "";
        for(int i:fileT.keySet())
        {
            Tuple tup = (Tuple)fileT.get(i);
            msg += i+" " + tup.getName() + '\n';
        }
        return msg;
    }

    public HashMap<Integer, T> getFT(){return (HashMap<Integer, T>) fileT; }
}
